cmd/can/main.go

```diff
@@ -3,91 +3,89 @@ package main
 import (
 	"flag"
 	"fmt"
+	"github.com/sasswart/gin-in-a-can/config"
+	"github.com/sasswart/gin-in-a-can/render"
 	"os"
-	"path"
 	"path/filepath"
 
-	"github.com/sasswart/gin-in-a-can/generator"
 	"github.com/sasswart/gin-in-a-can/openapi"
 	"github.com/spf13/viper"
 )
 
-type Config struct {
-	Generator         generator.Config
-	OutputPath        string
-	TemplateDirectory string
-	WorkingDirectory  string
-	ConfigFilePath    string
-}
-
 func main() {
-	config, err := loadConfig()
+	configData, err := loadConfig()
 	if err != nil {
 		fmt.Println(fmt.Errorf("loadConfig error: %w", err))
 		os.Exit(1)
 	}
 
-	fmt.Printf("Reading API specification from \"%s\"\n", absoluteOpenAPIFile(config))
+	fmt.Printf("Reading API specification from \"%s\"\n", absoluteOpenAPIFile(configData))
 	apiSpec, err := openapi.LoadOpenAPI(
-		absoluteOpenAPIFile(config),
+		absoluteOpenAPIFile(configData),
 	)
 	if err != nil {
 		fmt.Println(fmt.Errorf("openapi.LoadOpenAPI error: %w", err))
 		os.Exit(1)
 	}
 
-	generatorConfig := config.Generator.WithServer(*apiSpec)
-	for _, target := range []struct {
-		pkg      string
-		file     string
-		template string
-	}{
-		{"controller", "controller.go", "controller.tmpl"},
-		{"controller", "unimplemented.go", "unimplemented.tmpl"},
-		{"models", "models.go", "models.tmpl"},
-	} {
-		file, err := generator.Generate(generatorConfig, target.template)
-		if err != nil {
-			fmt.Println(err)
-		}
+	err = openapi.SetRenderer(apiSpec, openapi.GinRenderer{})
+	if err != nil {
+		fmt.Println(fmt.Errorf("openapi.SetRenderer error: %w", err))
+		os.Exit(1)
+	}
+
+	apiSpec.SetMetadata(map[string]string{
+		"package": "api",
```

Comment:
should this not be set from `can.yml:4`?

Reply:
fixed